### PR TITLE
[virt_autotest] fix virtual_network_utils issue on sles11sp4

### PR DESCRIPTION
**Pull request by guoxuguang**
Refer to  the following virtual_network_utils failures on sles11sp4, figure out that systemctl cmd do not work on 11sp4 from restart_libvirtd fun under virtual_network_utils pm.
gi-guest_developing-on-host_sles11sp4-kvm-dev
https://openqa.nue.suse.com/tests/4262311
gi-guest_developing-on-host_sles11sp4-xen-dev
https://openqa.nue.suse.com/tests/4262312

This PR fix this virtual_network_utils issue on sles11sp4 vm hosts

- Verification run: 
gi-guest_developing-on-host_sles11sp4-xen-dev
http://10.67.19.82/tests/564
gi-guest_developing-on-host-developing-xen
http://10.67.19.82/tests/566

### CODE REVIEW

**Comment by alice-suse** [comment]
LGTM.  Title is still WIP, not sure whether ready to be merged? @guoxuguang

**Comment by guoxuguang** [comment]
> LGTM. Title is still WIP, not sure whether ready to be merged? @guoxuguang
For now,  current code change work with gi-guest_developing-on-host_sles11sp4-xen-kvm
Working on verification run with gi-guest_developing-on-host_sles11sp4-xen-dev 
If there was not any error or comments, please help merger this PR.

**Comment by guoxuguang** [comment]
updated the latest Verification run as below:
gi-guest_developing-on-host_sles11sp4-kvm-dev
http://10.67.19.82/tests/524
gi-guest_developing-on-host_sles11sp4-xen-dev
http://10.67.19.82/tests/521
gi-guest_developing-on-host-developing-kvm
http://10.67.19.82/tests/522

**Comment by guoxuguang** [comment]
Code change and verification run almost done already. @alice-suse @waynechen55 @Julie-CAO would you mind help me review this PR. 
@alice-suse if there was not any comments, please help me merge this PR.

**Comment by guoxuguang** [comment]
@pdostal please help me make sure if this PR work on QAM environment.

**Comment by Julie-CAO** [comment]
LGTM

File: tests/virt_autotest/libvirt_routed_virtual_network.pm [hunk 1]
```diff
@@ -70,17 +71,23 @@ sub run_test {
...
+        assert_script_run("virsh attach-interface $guest.clone network vnet_routed_clone --model $model2 --mac $mac2 --live $affecter", 60);
+
+        #Wait for guests attached interface from virtual routed network
+        sleep 30;
+        my $net1 = (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen') && is_sle('=11-sp4')) ? 'netfront' : 'vnet_routed';
+        test_network_interface("$guest", mac => $mac1, gate => $gate1, target => $target1, net => $net1);
+        my $net2 = (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen') && is_sle('=11-sp4')) ? 'netfront' : 'vnet_routed_clone';
+        test_network_interface("$guest.clone", mac => $mac2, gate => $gate2, target => $target2, net => $net2);
+
+        my $exclusive = (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen') && is_sle('=11-sp4')) ? 'bridge --live --persistent' : 'network --current';
```
Comment (pdostal) [inline tests/virt_autotest/libvirt_routed_virtual_network.pm:88]: Why do you use `bridge` here? This should test two separated networks routed together.

File: tests/virt_autotest/libvirt_routed_virtual_network.pm [hunk 2]
```diff
@@ -70,17 +71,23 @@ sub run_test {
...
-
-        test_network_interface("$guest",       mac => $mac1, gate => $gate1, target => $target1, net => "vnet_routed");
-        test_network_interface("$guest.clone", mac => $mac2, gate => $gate2, target => $target2, net => "vnet_routed_clone");
-
-        assert_script_run("virsh detach-interface $guest network --mac $mac1 --current");
-        assert_script_run("virsh detach-interface $guest.clone network --mac $mac2 --current");
+        assert_script_run("virsh attach-interface $guest.clone network vnet_routed_clone --model $model2 --mac $mac2 --live $affecter", 60);
+
+        #Wait for guests attached interface from virtual routed network
+        sleep 30;
```
Comment (pdostal) [inline tests/virt_autotest/libvirt_routed_virtual_network.pm:82]: This is not needed and we should avoid `sleep` in general.

File: tests/virt_autotest/libvirt_virtual_network_init.pm [hunk 3]
```diff
@@ -53,28 +54,22 @@ sub run_test {
...
         assert_script_run("virsh dumpxml $guest > /tmp/$guest.xml");
         upload_logs "/tmp/$guest.xml";
+        assert_script_run("rm -rf /tmp/$guest.xml");
         #Start installed Guests
         assert_script_run("virsh start $guest", 60);
         #Wait for forceful boot up guests
         sleep 60;
         save_guest_ip($guest, name => "br123");
-        exec_and_insert_password("ssh-copy-id -o StrictHostKeyChecking=no -f root\@$guest");
+        my $mode = is_sle('=11-sp4') ? '' : -f;
```
Comment (pdostal) [inline tests/virt_autotest/libvirt_virtual_network_init.pm:63]:
```suggestion
        my $mode = is_sle('=11-sp4') ? '' : '-f';
```

File: tests/virt_autotest/libvirt_routed_virtual_network.pm (hunk 1 above)
Comment (guoxuguang) [reply tests/virt_autotest/libvirt_routed_virtual_network.pm:88]: After attach virtual interface from our virtual network(e.g vnet_routed_clone), there was not virtual network name as SOURCE on sles11sp4 xen host, just only TYPE - bridge and mode - netfront from domiflist output. 
Such as:
virsh domiflist sles-15-sp2
IN  TYPE MODE MAC
--------------------------------------------
vif8 bridge netfront XX.XX.XX.XX

So, have to use with "bridge" to find out guest mac under save_guest_ip func on s11sp4 xen host here.

File: tests/virt_autotest/libvirt_routed_virtual_network.pm (hunk 2 above)
Comment (guoxuguang) [reply tests/virt_autotest/libvirt_routed_virtual_network.pm:82]: Good point. I agree that. 
But, I had confirm that after attached virtual interface form **routed network test** to guest system on sles kvm (sles15sp2) host, and then to trigger test_network_interface() immediately, found out some TIMEOUT issue to connect guest via ssh used with ip from routed network. 
please refer to the following link for more details
https://openqa.nue.suse.com/tests/4263937#step/libvirt_routed_virtual_network/105
So, we'd better to add this sleep operation after attached rtual interface form **routed network test** to guest system, and then trigger trigger test_network_interface().

File: tests/virt_autotest/libvirt_virtual_network_init.pm (hunk 3 above)
Comment (guoxuguang) [reply tests/virt_autotest/libvirt_virtual_network_init.pm]: Had updated this code changed.

**Comment by guoxuguang** [comment]
> Hello, from my point of view there is now a lot of exceptions and the code is becoming more and more messy. Please give me some time, we are currently busy with testing updates but I will test your PR as soon as possible.

Great, thank you so much for your great help. 
Feel free to let me know if there was any comments and suggestion here

**Comment by guoxuguang** [comment]
Quick updated, refer to the latest results of osd build 203.1, there were the following analyzed results about libvirt_routed_virtual_network failures:

- ssh connection rejected with guest system after finished save_guest_ip fun during libvirt_routed_virtual_network

ROOT CAUSE:
After attached routed virtual interface to guest system (i.e. sles15sp2), guest system need time to prepare ROUTED network environment (included ip address, ssh connection and so on) on the different version sles VM host, after manually routed network most 60 times, sum up the following the info about guest system required time to prepare ROUTED network on the different version sles VM host(kvm|xen):
guest sles15sp2 host: sles11sp4 -> 30-60 sec
guest sles15sp2 host: sles12+    ->  20-30 sec
guest sles 15sp2 host:sles15sp2 -> 10-20 sec 
SOLUTION:
after finished after finished save_guest_ip fun, add guest SSH service checked (timeout=60)

- SSH TIMEOUT to configure the network interface to use DHCP configuration under test_network_interface fun 

ROOT CAUSE:
After configured the network interface to use DHCP configuration just only with BOOTPROTO=dhcp option to guest system based on routed network,  routed virtual interface was DOWN status without ip address under guest system.

found that our guest system (i.e. sles15sp2) required STARTMODE option during routed virtual network

SOLUTION
During Configure the network interface to use DHCP configuration, add STARTMODE option to guest system based on routed network.

**Pull request merged**